concurrent/futures/thread.py

```diff
@@ -67,7 +67,9 @@ def run(self):
         else:
             self.future.set_result(result)
 
-def _worker(executor_reference, work_queue):
+def _worker(executor_reference, work_queue, initializer, initargs):
+    if initializer is not None:
+        initializer(*initargs)
```

Comment:
This looks quite different from upstream code. For one, exceptions are not handled.

Reply:
Ah, I wasn't aware this was a straight copy of upstream. I'll do that now

Reply:
It was originally copied from upstream obviously but modified for Python 2. I try to stick as close to upstream as reasonably possible.

Reply:
Okay, then I've copied the Python3 version of this, and pulled in the Broken* exceptions too

Reply:
Good. If there are any tests upstream that cover the failure of the initializer, I'd like those added too.